lib/WTSI/NPG/HTS/PacBio/Sequel/AnalysisFastaManager.pm

```diff
@@ -0,0 +1,192 @@
+
+=cut
+
+sub make_loadable_files {
+  my ($self) = shift;
+
+  my $output_dir  = $self->has_output_dir ? $self->output_dir :
+    $self->runfolder_path;
+
+  my $is_success = 0;
+  try {
+    my $fasta_path = catdir($self->analysis_path,$ISODEMUX);
+    if (! -e $fasta_path || ! -d $fasta_path) {
+      $self->info('Currently only handling multiplexed data');
+    } else {
+      my @fa  = $self->list_directory
+        ($fasta_path, filter => 'fasta' .q[$]);
+      my @xml = $self->list_directory
+        ($self->runfolder_path, filter => 'xml' .q[$]);
+
+      if (scalar @fa < 2 || scalar @fa != (scalar @xml * 2)) {
+        $self->logcroak('Expect >= 2 fasta files and 2 fasta per xml');
+      }
+      $self->_process_deplexed_data(\@fa,\@xml);
+      $is_success = 1;
+    }
+  } catch {
+    my @stack = split /\n/msx;   # Chop up the stack trace
+    $self->logcroak(pop @stack); # Use a shortened error message
+  };
+
+  if (!$is_success) {
+    my @gz = $self->list_directory
+      ($self->runfolder_path, filter => $FASUFFIX .q[$]);
+    foreach my $file (@gz) {
+      unlink $file or $self->logcroak('Failed to remove: ', $file);
+    }
+  }
+  return $is_success;
+}
+
+sub _process_deplexed_data {
+
+  my ($self, $fafiles, $xmlfiles) = @_;
+
+  my %sample2barcode;
+  foreach my $xmlfile (@{$xmlfiles}) {
+    my $prefix = 'pbsample:';
+    my $dom = XML::LibXML->new->parse_file($xmlfile);
+
+    my @biosample = $dom->getElementsByTagName($prefix . 'BioSample')->[1];
+    my $sample = $biosample[0]->getAttribute('Name');
+
+    my @dnabarcode = $dom->getElementsByTagName($prefix . 'DNABarcode')->[0];
+    my $barcode = $dnabarcode[0]->getAttribute('Name');
+
+    if(!$barcode || !$sample ){
+      $self->logcroak('XML file without sample or barcode info: ', $xmlfile);
+    } else {
+      $sample =~ s/BioSample_//smx;
+      $sample2barcode{$sample} = $barcode;
+    }
+  }
+
+  foreach my $fafile (@{$fafiles}) {
+    my($type,$sample);
+    if ($fafile =~ /(\wq_transcripts) [-] (\d+) [.] fasta/smx) {
+      ($type,$sample) = ($1,$2);
+    } else {
+      $self->logcroak('Fasta file found with unexpected name: ', $fafile);
+    }
+
+    my $bc = defined $sample2barcode{$sample} ? $sample2barcode{$sample} :
+      $self->logcroak('Fasta file found with no barcode conversion: ', $fafile);
+
+    my $faout = catfile($self->runfolder_path,
+      $self->meta_data->movie_name. q[.]. $bc .q[.]. $type .q[.]. $FASUFFIX);
+
+    if (! -f $faout) {
+      my $fa = IO::File->new($fafile, q[<])
+        or $self->logcroak('Could not open: ', $fafile);
+      my $gz = IO::Compress::Gzip->new($faout)
+        or $self->logcroak('Could not write to: ', $faout);
+
+      $self->info('Writing compressed fasta to: ', $faout);
+      if (defined $fa) {
+        while (my $line = <$fa>) {
+          if ( $line =~ /^>/smx ){
+            my @l = split /\s+/smx, $line;
+            $gz->write($l[0]."\n");
+          } else {
+            $gz->write($line);
```

Comment:
I guess the explicit streaming here because the fasta header line needs to be re-written? It would be helpful to have a comment about what it's keeping/removing from the header.